Rubberduck.Parsing/VBA/ParseCoordinator.cs

```diff
@@ -86,7 +97,35 @@ private void ReparseRequested(object sender, EventArgs e)
                 token = _currentCancellationTokenSource.Token;
             }
 
+            BeginParse(sender,token);
+        }
+
+        public void SuspendRequested(object sender, RubberduckStatusSuspendParserEventArgs e)
+        {
+            lock (_cancellationSyncObject)
+            {
+                if (State.Status != ParserState.Ready)
+                {
+                    throw new InvalidOperationException(
```

Comment:
Who guarantees that you are in this state when you are entering this method? The state can get changed from another thread.

Reply:
I need to revisit that now that the suspension request is a basically a firing an event from RPS to the PC. The first iteration was meant to be synchronous check because I wanted to avoid the additional complexity of queuing a suspension action since the suspension itself already queues the parse requests.